src/include/catalog/postgres/pg_type_impl.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <string>
+
+#include "catalog/postgres/pg_type.h"
+#include "common/managed_pointer.h"
+#include "storage/projected_row.h"
+#include "storage/storage_defs.h"
+
+namespace noisepage::storage {
+class RecoveryManager;
+class SqlTable;
+
+namespace index {
+class Index;
+}  // namespace index
+}  // namespace noisepage::storage
+
+namespace noisepage::transaction {
+class TransactionContext;
+}  // namespace noisepage::transaction
+
+namespace noisepage::catalog::postgres {
+class Builder;
+
+/** The NoisePage version of pg_type. */
+class PgTypeImpl {
+ public:
+  /**
+   * Prepare to create pg_type.
+   * Does NOT create anything until the relevant bootstrap functions are called.
+   *
+   * @param db_oid          The OID of the database that pg_type should be created in.
+   */
+  explicit PgTypeImpl(db_oid_t db_oid);
+
+  /** Bootstrap the projected row initializers for pg_type. */
+  void BootstrapPRIs();
+
+  /**
+   * Bootstrap:
+   *    pg_type
+   *    pg_type_oid_index
+   *    pg_type_name_index
+   *    pg_type_namespace_index
```

Comment:
Add a description of dependency relations with other tables.

Reply:
wip